pandas/tests/test_frame.py

```diff
@@ -14751,6 +14751,16 @@ def test_query_builtin(self):
         result = df.query('sin > 5', engine=engine, parser=parser)
         tm.assert_frame_equal(expected, result)
 
+    def test_query_with_assign_statement(self):
+        df = DataFrame({'a': [1, 2, 3], 'b': ['a', 'b', 'c']})
+        a_before = df['a'].copy()
+        self.assertRaisesRegexp(
+            NotImplementedError, "'Assign' nodes are not implemented",
```

Comment:
the ``NotImplementedError`` is fine. I agree ideally should be ``SyntaxError`` but that's a separate change (welcome to make an issue if so desired)